context.go

```diff
@@ -336,38 +336,62 @@ func (c *Context) Rem(d, x, y *Decimal) (Condition, error) {
 }
 
 // Sqrt sets d to the square root of x.
+// Sqrt uses the Babylonian method for computing the square root, which uses
+// O(log p) steps for p digits of precision. We follow the algorithm in
+// Properly Rounded Variable Precision Square Root by T. E. Hull and A. Abrham,
+// ACM Transactions on Mathematical Software, Vol 11 #3, pp229–237, ACM, September 1985.
 func (c *Context) Sqrt(d, x *Decimal) (Condition, error) {
-	// See: Properly Rounded Variable Precision Square Root by T. E. Hull
-	// and A. Abrham, ACM Transactions on Mathematical Software, Vol 11 #3,
-	// pp229–237, ACM, September 1985.
 
 	switch x.Sign() {
 	case -1:
-		res := InvalidOperation
-		return c.goError(res)
+		return c.goError(InvalidOperation)
 	case 0:
 		d.SetCoefficient(0)
 		d.Exponent = 0
 		return 0, nil
 	}
-
-	// Use same precision as in decNumber.
-	workp := c.Precision + 1
-	if nd := uint32(x.NumDigits()); workp < nd {
-		workp = nd
+	// x > 0.
+
+	// TODO(mjibson): Also it would be good if we could keep the number of
+	// variables in scope low. This refactoring doesn't quite work, but perhaps
+	// you can refactor the calls below to use nc.Precision instead of keeping
+	// workp in scope throughout.
```

Comment:
I'm not sure how to do this without an extra variable because nc.Precision needs to be set to workp later down in the function, so we have to store it somewhere.